diff -Naur openldap-2.4.56/configure openldap-2.4.56-changelog/configure
--- openldap-2.4.56/configure	2020-11-10 19:22:13.000000000 +0100
+++ openldap-2.4.56-changelog/configure	2020-12-28 00:30:41.284444497 +0100
@@ -988,6 +988,7 @@
 enable_xxslapoverlays
 enable_overlays
 enable_accesslog
+enable_changelog
 enable_auditlog
 enable_collect
 enable_constraint
@@ -1688,6 +1689,7 @@
 SLAPD Overlay Options:
     --enable-overlays	  enable all available overlays no|yes|mod
     --enable-accesslog	  In-Directory Access Logging overlay no|yes|mod [no]
+    --enable-changelog    Changelog implementation overlay no|yes|mod [no]
     --enable-auditlog	  Audit Logging overlay no|yes|mod [no]
     --enable-collect	  Collect overlay no|yes|mod [no]
     --enable-constraint	  Attribute Constraint overlay no|yes|mod [no]
@@ -4403,6 +4405,7 @@
 # end --enable-sql
 
 Overlays="accesslog \
+        changelog \
 	auditlog \
 	collect \
 	constraint \
@@ -4470,6 +4473,28 @@
 
 # end --enable-accesslog
 
+# OpenLDAP --enable-changelog
+
+       # Check whether --enable-changelog was given.
+if test "${enable_changelog+set}" = set; then :
+  enableval=$enable_changelog;
+        ol_arg=invalid
+        for ol_val in no yes mod ; do
+                if test "$enableval" = "$ol_val" ; then
+                        ol_arg="$ol_val"
+                fi
+        done
+        if test "$ol_arg" = "invalid" ; then
+                as_fn_error "bad value $enableval for --enable-changelog" "$LINENO" 5
+        fi
+        ol_enable_changelog="$ol_arg"
+
+else
+        ol_enable_changelog=${ol_enable_overlays:-no}
+fi
+
+# end --enable-changelog
+
 # OpenLDAP --enable-auditlog
 
 	# Check whether --enable-auditlog was given.
@@ -24399,6 +24424,23 @@
 
 fi
 
+if test "$ol_enable_changelog" != no ; then
+        BUILD_CHANGELOG=$ol_enable_changelog
+        if test "$ol_enable_changelog" = mod ; then
+                MFLAG=SLAPD_MOD_DYNAMIC
+                SLAPD_DYNAMIC_OVERLAYS="$SLAPD_DYNAMIC_OVERLAYS changelog.la"
+        else
+                MFLAG=SLAPD_MOD_STATIC
+                SLAPD_STATIC_OVERLAYS="$SLAPD_STATIC_OVERLAYS changelog.o"
+        fi
+
+cat >>confdefs.h <<_ACEOF
+#define SLAPD_OVER_CHANGELOG $MFLAG
+_ACEOF
+
+fi
+
+
 if test "$ol_enable_auditlog" != no ; then
 	BUILD_AUDITLOG=$ol_enable_auditlog
 	if test "$ol_enable_auditlog" = mod ; then
diff -Naur openldap-2.4.56/configure.in openldap-2.4.56-changelog/configure.in
--- openldap-2.4.56/configure.in	2020-11-10 19:22:13.000000000 +0100
+++ openldap-2.4.56-changelog/configure.in	2020-12-28 00:33:34.108227662 +0100
@@ -337,6 +337,7 @@
 dnl ----------------------------------------------------------------
 dnl SLAPD Overlay Options
 Overlays="accesslog \
+        changelog \
 	auditlog \
 	collect \
 	constraint \
@@ -364,6 +365,8 @@
 	--, [no yes mod])dnl
 OL_ARG_ENABLE(accesslog,[    --enable-accesslog	  In-Directory Access Logging overlay],
 	no, [no yes mod], ol_enable_overlays)
+OL_ARG_ENABLE(changelog,[    --enable-changelog   Changelog implementation overlay],
+        no, [no yes mod], ol_enable_overlays)
 OL_ARG_ENABLE(auditlog,[    --enable-auditlog	  Audit Logging overlay],
 	no, [no yes mod], ol_enable_overlays)
 OL_ARG_ENABLE(collect,[    --enable-collect	  Collect overlay],
@@ -2830,6 +2833,18 @@
 	AC_DEFINE_UNQUOTED(SLAPD_OVER_ACCESSLOG,$MFLAG,[define for In-Directory Access Logging overlay])
 fi
 
+if test "$ol_enable_changelog" != no ; then
+        BUILD_CHANGELOG=$ol_enable_changelog
+        if test "$ol_enable_changelog" = mod ; then
+                MFLAG=SLAPD_MOD_DYNAMIC
+                SLAPD_DYNAMIC_OVERLAYS="$SLAPD_DYNAMIC_OVERLAYS changelog.la"
+        else
+                MFLAG=SLAPD_MOD_STATIC
+                SLAPD_STATIC_OVERLAYS="$SLAPD_STATIC_OVERLAYS changelog.o"
+        fi
+        AC_DEFINE_UNQUOTED(SLAPD_OVER_ACCESSLOG,$MFLAG,[define for Changelog implementation overlay])
+fi
+
 if test "$ol_enable_auditlog" != no ; then
 	BUILD_AUDITLOG=$ol_enable_auditlog
 	if test "$ol_enable_auditlog" = mod ; then
diff -Naur openldap-2.4.56/include/portable.hin openldap-2.4.56-changelog/include/portable.hin
--- openldap-2.4.56/include/portable.hin	2020-11-10 19:22:13.000000000 +0100
+++ openldap-2.4.56-changelog/include/portable.hin	2020-12-28 00:34:22.224497235 +0100
@@ -963,6 +963,9 @@
 /* define for In-Directory Access Logging overlay */
 #undef SLAPD_OVER_ACCESSLOG
 
+/* define for Changelog implementation overlay */
+#undef SLAPD_OVER_CHANGELOG
+
 /* define for Audit Logging overlay */
 #undef SLAPD_OVER_AUDITLOG
 
